src/vabackend.c

```diff
@@ -675,19 +823,61 @@ static VAStatus nvQueryConfigAttributes(
     *profile = cfg->profile;
     *entrypoint = cfg->entrypoint;
     int i = 0;
+    attrib_list[i].value = 0;
     attrib_list[i].type = VAConfigAttribRTFormat;
-    attrib_list[i].value = VA_RT_FORMAT_YUV420;
-    if (drv->supports16BitSurface) {
-        switch(cfg->profile) {
-        case VAProfileHEVCMain12:
-        case VAProfileVP9Profile2:
-            attrib_list[i].value |= VA_RT_FORMAT_YUV420_12;
-            // Fall-through
-        case VAProfileHEVCMain10:
-        case VAProfileAV1Profile0:
-            attrib_list[i].value |= VA_RT_FORMAT_YUV420_10;
-            break;
-    }}
+    switch (cfg->profile)
+    {
+    case VAProfileHEVCMain12:
+    case VAProfileVP9Profile2:
+        attrib_list[i].value = VA_RT_FORMAT_YUV420_12;
```

Comment:
You've missed the `|=`, which changes what RT formats will be returned. It seems to be missing from the rest of them too.